esp/esdllib/esdl_script.cpp

```diff
@@ -2883,6 +2883,119 @@ class CEsdlTransformOperationDelay : public CEsdlTransformOperationWithoutChildr
     }
 };
 
+class CEsdlTransformOperationUpdateMaskingContext : public CEsdlTransformOperationBase
+{
+    class Update : public CEsdlTransformOperationWithoutChildren
```

Comment:
Do updates really need to be operations at all?  You just seem to be iterating over them and calling apply.. if there is no additional operation like behavior you could consider simplifying the class.

Reply:
Making each type of update an operation provides default behaviors like supporting `trace` (and `optional`) in the markup, as well as standard error reporting for invalid markup. I modeled it after the `mysql` operation and its `bind` children which are implemented as operations even though they are not used that way.

The `Update` base class, as opposed to an interface, satisfies the shared requirements to be an operation, simplifying each subclass. It could be replaced by an interface implemented by each subclass if you think it would be cleaner.

Reply:
Yes, ok that's fine.  I just noticed it only seemed to be overriding the toDBGLog().